gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -461,7 +467,14 @@ public SubscriptionEntity process(ProcessSubscriptionEntity processSubscription,
 
             if (plan.getSecurity() == PlanSecurityType.API_KEY &&
                     subscription.getStatus() == Subscription.Status.ACCEPTED) {
-                apiKeyService.generate(subscription.getId());
+                if (StringUtils.isNotEmpty(processSubscription.getCustomApiKey())) {
+                    if (apiKeyService.exists(processSubscription.getCustomApiKey())) {
```

Comment:
What if customApiKey is an old apikey that expired . `apiKeyServicee.exists()` will return `true` but the apiKey is not activated.
IMHO you should add a test or change the thrown exception.

Reply:
as discussed, the api-key value is its id. so an api key is not reusable